packages/main/cypress/specs/RadioButton.cy.tsx

```diff
@@ -0,0 +1,497 @@
+
+		cy.get("#groupRbReadOnly")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("have.attr", "aria-disabled");
+	});
+});
+
+describe("RadioButton general interaction", () => {
+	it("tests change event", () => {
+		const changeEventStub = cy.stub();
+		cy.mount(<RadioButton id="rb1" onChange={changeEventStub}></RadioButton>);
+
+		cy.get("#rb1")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb1").realClick();
+		cy.wrap(changeEventStub)
+			.should("have.been.called");
+
+		cy.get("#rb1").realClick();
+		cy.wrap(changeEventStub)
+			.should("have.been.calledOnce");
+	});
+
+	it("tests change event upon ENTER", () => {
+		const changeEventStub = cy.stub();
+		cy.mount(
+			<>
+				<RadioButton id="rb1"></RadioButton>
+				<RadioButton id="rb2" text="Option B" onChange={changeEventStub}></RadioButton>
+			</>);
+
+		cy.get("#rb1")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb1").realClick();
+		cy.get("#rb1").realPress("Tab");
+
+		cy.get("#rb2")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb2").realPress("Enter");
+		cy.wrap(changeEventStub)
+			.should("have.been.called");
+
+		cy.get("#rb2").realPress("Enter");
+		cy.wrap(changeEventStub)
+			.should("have.been.calledOnce");
+	});
+
+	it("tests change event upon SPACE", () => {
+		const changeEventStub = cy.stub();
+		cy.mount(
+			<>
+				<RadioButton id="rb2" text="Option B"></RadioButton>
+				<RadioButton id="rb3" wrappingType="Normal" text="Option C" onChange={changeEventStub}></RadioButton>
+			</>);
+
+		cy.get("#rb2")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb2").realClick();
+		cy.get("#rb2").realPress("Tab");
+
+		cy.get("#rb3")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb3").realPress("Space");
+		cy.wrap(changeEventStub)
+			.should("have.been.called");
+
+		cy.get("#rb3").realPress("Space");
+		cy.wrap(changeEventStub)
+			.should("have.been.calledOnce");
+	});
+
+	it("tests change event not fired, when disabled", () => {
+		const changeEventStub = cy.stub();
+		cy.mount(<RadioButton id="rb4" disabled text="Option D" onChange={changeEventStub}></RadioButton>);
+
+		cy.get("#rb4")
+			.shadow()
+			.find(".ui5-radio-root")
+			.should("exist");
+
+		cy.get("#rb4").realClick();
+		cy.get("#rb4").realPress("Space");
+		cy.get("#rb4").realPress("Enter");
+
+		cy.wrap(changeEventStub)
+			.should("not.have.been.called");
+	});
+
+	it("tests radio buttons selection within group with ARROW-RIGHT key", () => {
+		cy.mount(
+			<>
+				<Input id="tabField"></Input>
+				<RadioButton id="groupRb1" name="a" wrappingType="None" text="Option A long long should shrink long long text text text text text text text text"></RadioButton>
+				<RadioButton id="groupRb3" name="a" text="Option D"></RadioButton>
+				<RadioButton id="groupRbReadOnly" name="a" readonly text="Option E"></RadioButton>
```

Comment:
add disabled radio button as well and probably you can combine with:
- tests radio buttons selection within group with ARROW-LEFT key